r2i/tflite/frameworkfactory.h

```diff
@@ -13,6 +13,7 @@
 #define R2I_TFLITE_FRAMEWORK_FACTORY_H
 
 #include <r2i/iframeworkfactory.h>
+#include <tensorflow/lite/version.h>
```

Comment:
Notice the problem here: by adding the include to the header when unneeded, you are forcing everyone who includes this header to an extra dependency. By keeping it in the source file you encapsulate this dependency! 

My rule of thumb is:

1. place the include in the source file
2. if it is needed in the header, evaluate why. Can this dependency be avoided? (By using polymorphism, abstract data types, encapsulation, etc...)
3. If not, move it to the header.